src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationPatternMatchingForSwitch.java

```diff
@@ -0,0 +1,86 @@
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * tabWidth = 4                                                             //indent:1 exp:1
+ * throwsIndent = 8                                                         //indent:1 exp:1
+ * forceStrictCondition = true                                             //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+public class InputIndentationPatternMatchingForSwitch {                     //indent:0 exp:0
+    record Point(int x, int y) { }                                           //indent:4 exp:4
+    record square(Point upperLeft, Point lowerRight) { }                    //indent:4 exp:4
+
+    public void test1() {   //indent:4 exp:4
+        Point p = new Point(1, 2); //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case Point(int x, int y) //indent:12 exp:12
+            when x > 0 && y > 0 -> System.out.println(x + y); //indent:12 exp:16 warn
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case Point(int x, int y) //indent:12 exp:12
+                when x > 0 && y > 0 -> System.out.println(x + y); //indent:16 exp:16
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+    } //indent:4 exp:4
+
+    public void test2() { //indent:4 exp:4
+        Point p = new Point(1, 2); //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case    //indent:12 exp:12
+                    Point(int x, int y) //indent:20 exp:20
+                when x > 0 && y > 0 -> System.out.println(x + y); //indent:16 exp:16
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case    //indent:12 exp:12
+                    Point(int x, int y) //indent:20 exp:20
+                    when x > 0 && y > 0 -> System.out.println(x + y); //indent:20 exp:20
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case    //indent:12 exp:12
+                    Point(int x, int y) //indent:20 exp:20
+                            when x > 0 && y > 0 -> System.out.println(x + y); //indent:28 exp:28
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
+        switch (p) { //indent:8 exp:8
+            case    //indent:12 exp:12
+                    Point(int x, int y) //indent:20 exp:20
+            when x > 0 && y > 0 -> System.out.println(x + y); //indent:12 exp:16 warn
+            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
+        } //indent:8 exp:8
```

Comment:
Add an input like: (indentation is intentional)
````
        switch (p) { //indent:8 exp:8
            case    //indent:12 exp:12
Point(int x, int y)
when
x > 0 && y > 0
->
System.out.println(x + y);
            case Point(int x, int y) -> System.out.println("error"); //indent:12 exp:12
        } //indent:8 exp:8
````

Reply:
> Just to be clear, we are only validating `LITERAL_WHEN` and things around it, correct?

Yes



> Add an input like: (indentation is intentional)

done.

> System.out.println(x + y);

this is weird indent:0 expected:0

Reply:
> this is weird indent:0 expected:0

Exactly. :)

This is a bug. Most likely this line is not being validated.

You will only see issues with this check if you do bad indentation. You other examples were "too good".

After talking with @nrmancuso , move this to a new issue. We need to continue opening other issues and will probably return to indentation later.

Reply:
@mahfouz72 Is this done?

Reply:
https://github.com/checkstyle/checkstyle/issues/15098